File: Java/labs/lab15_Command/task_2_1/src/TextEditor.java
public class TextEditor {
    public static void main(String[] args) {

        Document document = new Document("blank");

        Button saveButton = new Button();
        Button printButton = new Button();

        MenuItem saveMenuItem = new MenuItem();
        MenuItem printMenuItem = new MenuItem();
        MenuItem showInfoMenuItem = new MenuItem();

        Shortcut saveShortcut = new Shortcut();
        Shortcut printShortcut = new Shortcut();

        /////////////////////////////////////////////
        // Testing buttons, menu items, shortcuts...
        //////////////////////////////////////////////
        saveButton.click();
        printButton.click();

        saveMenuItem.select();
        printMenuItem.select();
        showInfoMenuItem.select();

        saveShortcut.press();
        printShortcut.press();
    }
}

File: Cpp/lab15_Command/task_2_1/src/text_editor.cpp
#include "text_editor.hpp"


TextEditor::TextEditor() :
  saveButton(Button()),
  printButton(Button()),

  saveMenuItem(MenuItem()),
  printMenuItem(MenuItem()),
  showInfoMenuItem(MenuItem()),

  saveShortcut(Shortcut()),
  printShortcut(Shortcut())
{}

TextEditor::~TextEditor()
{
  if (this->document != nullptr)
    delete this->document;
}

void TextEditor::newDocument(std::string const& name)
{
  this->document = new Document(name);
}

void TextEditor::clickSaveButton()
{
  this->saveButton.click();
}

void TextEditor::clickPrintButton()
{
  this->saveMenuItem.select();
}

void TextEditor::selectPrintMenuItem()
{
  this->printMenuItem.select();
}

void TextEditor::selectSaveMenuItem()
{
  this->saveMenuItem.select();
}

void TextEditor::selectShowInfoMenuItem()
{
  this->showInfoMenuItem.select();
}

void TextEditor::pressSaveShortcut()
{
  this->saveShortcut.press();
}

void TextEditor::pressPrintShortcut()
{
  this->printShortcut.press();
}
